src/schematools/types.py

```diff
@@ -536,43 +564,45 @@ def identifier(self) -> List[str]:
         # Convert identifier to a list, to be backwards compatible with older schemas
         if not isinstance(identifier, list):
             identifier = [identifier]
-        return identifier
+        return list(identifier)  # mypy pleaser
 
     @property
     def has_compound_key(self) -> bool:
-        if isinstance(self.identifier, str):
-            return False
+        # Mypy bug that has been resolved but not merged
+        # https://github.com/python/mypy/issues/9907
+        if isinstance(self.identifier, str):  # type: ignore[unreachable]
+            return False  # type: ignore[unreachable]
         return len(self.identifier) > 1
 
-    def validate(self, row: Dict[str, Any]):
+    def validate(self, row: Dict[str, Any]) -> None:
         """Validate a record against the schema."""
         jsonschema.validate(row, self.data["schema"])
 
-    def _resolve(self, ref):
+    def _resolve(self, ref: str) -> jsonschema.RefResolver:
         """Resolve the actual data type of a remote URI reference."""
         return jsonschema.RefResolver(ref, referrer=self)
 
     @property
-    def has_parent_table(self):
+    def has_parent_table(self) -> bool:
         return "parentTableID" in self["schema"]
 
     @property
-    def filters(self):
+    def filters(self) -> Dict[str, Dict[str, str]]:
         """Fetch list of additional filters"""
-        return self["schema"].get("additionalFilters", {})
+        return dict(self["schema"].get("additionalFilters", {}))
 
     @property
-    def relations(self):
+    def relations(self) -> Dict[Any, Any]:
```

Comment:
I am not so familiar with relations. However `Any` for the key seems especially broad. I assume it can be `str` if I read the docstring correctly (_dictionary of names ..._)

Regarding the values. Does it resolve relations to actual table definitions? If so the return type can be `Dict[str, DatasetTableSchema]`.

If on the other hand it simple returns a JSON blob, you could turn it into `Dict[str, Json]` using the `Json` definition in https://github.com/Amsterdam/schema-tools/pull/159/files#r653344045

Reply:
Has always a fixed structure, changed it to `Dict[str, Dict[str, str]]`.

Reply:
You might consider making it an alias:
```python
Relations = Dict[str, Dict[str, str]]
...
def relations(self) -> Relations:
    ....
```
This way a future more strict (dataclass/Pydantic) based type can easily be introduced by search and replacing `Relations` with `<new-fancy-name-for-new-type>` and addressing the issues MyPy then finds.